app/src/main/java/com/example/androidthings/assistant/LedBlinkThread.java

```diff
@@ -0,0 +1,65 @@
+package com.example.androidthings.assistant;
+
+import com.google.android.things.pio.Gpio;
+
+import java.io.IOException;
+import java.util.Random;
+
+public class LedBlinkThread extends Thread {
+    private final Gpio mLed;
+    private final Random mRandom;
+    private boolean mBlinking = false;
+    private boolean mClose = false;
+
+    public LedBlinkThread(Gpio led) {
+        mLed = led;
+        mRandom = new Random();
+    }
+
+    public void setBlinking(boolean b) {
+        if (mClose) {
+            return;
+        }
+        if (mBlinking != b) {
+            mBlinking = b;
+            synchronized (this) {
+                notify();
+            }
+        }
+    }
+
+    public void close() {
+        mClose = true;
+        mBlinking = false;
+        synchronized (this) {
+            notify();
+        }
+    }
+
+    @Override
+    public void run() {
+        do {
+            synchronized (this) {
+                try {
+                    wait();
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+            while (mBlinking) {
+                mBlinking = false;
+                try {
+                    mLed.setValue(false);
+                    sleep(150+mRandom.nextInt(100));
```

Comment:
I'm not sure if random blinking delays are ideal.

Reply:
It makes it feel a bit more dynamic, IMHO if the interval is always the same it feels too static

Reply:
Can you provide a short video/gif of each one as a comparison? I'm not convinced that the random blinking is better. Consistency is not bad. This project is meant to be a sample and not a highly polished and complex project.

Reply:
I just changed it to a constant